pkg/entities/ie_value.go

```diff
@@ -10,2365 +9,636 @@ type InfoElementWithValue interface {
 	// TODO: Handle error to make it more robust if it is called prior to AddInfoElement.
 	GetInfoElement() *InfoElement
 	AddInfoElement(infoElement *InfoElement)
-	GetUnsigned8Value() (uint8, error)
-	GetUnsigned16Value() (uint16, error)
-	GetUnsigned32Value() (uint32, error)
-	GetUnsigned64Value() (uint64, error)
-	GetSigned8Value() (int8, error)
-	GetSigned16Value() (int16, error)
-	GetSigned32Value() (int32, error)
-	GetSigned64Value() (int64, error)
-	GetFloat32Value() (float32, error)
-	GetFloat64Value() (float64, error)
-	GetBooleanValue() (bool, error)
-	GetMacAddressValue() (net.HardwareAddr, error)
-	GetStringValue() (string, error)
-	GetIPAddressValue() (net.IP, error)
-	SetUnsigned8Value(val uint8) error
-	SetUnsigned16Value(val uint16) error
-	SetUnsigned32Value(val uint32) error
-	SetUnsigned64Value(val uint64) error
-	SetSigned8Value(val int8) error
-	SetSigned16Value(val int16) error
-	SetSigned32Value(val int32) error
-	SetSigned64Value(val int64) error
-	SetFloat32Value(val float32) error
-	SetFloat64Value(val float64) error
-	SetBooleanValue(val bool) error
-	SetMacAddressValue(val net.HardwareAddr) error
-	SetStringValue(val string) error
-	SetIPAddressValue(val net.IP) error
+	GetUnsigned8Value() uint8
+	GetUnsigned16Value() uint16
+	GetUnsigned32Value() uint32
+	GetUnsigned64Value() uint64
+	GetSigned8Value() int8
+	GetSigned16Value() int16
+	GetSigned32Value() int32
+	GetSigned64Value() int64
+	GetFloat32Value() float32
+	GetFloat64Value() float64
+	GetBooleanValue() bool
+	GetMacAddressValue() net.HardwareAddr
+	GetStringValue() string
+	GetIPAddressValue() net.IP
+	SetUnsigned8Value(val uint8)
+	SetUnsigned16Value(val uint16)
+	SetUnsigned32Value(val uint32)
+	SetUnsigned64Value(val uint64)
+	SetSigned8Value(val int8)
+	SetSigned16Value(val int16)
+	SetSigned32Value(val int32)
+	SetSigned64Value(val int64)
+	SetFloat32Value(val float32)
+	SetFloat64Value(val float64)
+	SetBooleanValue(val bool)
+	SetMacAddressValue(val net.HardwareAddr)
+	SetStringValue(val string)
+	SetIPAddressValue(val net.IP)
 	IsValueEmpty() bool
 	GetLength() int
 	ResetValue()
 }
 
-type Unsigned8InfoElement struct {
-	value   uint8
+type BaseInfoElement struct {
```

Comment:
Is it necessary to make this type public? I didn't see it is used outside this package.

Reply:
Not needed. Modified.